lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
Presently, I'm not convinced this isn't an application concern until formally defined by JSONAPI. I prefer explicit over implicit behaviors. I think the empty meta approach requires clients to be responsible for knowing that empty meta implies no relationship data was available **_without telling them how to go get more_**. That's why I probably would solve this by not providing `data` in relationships (omitting the data key) and providing a `link`. That would convey a relationship is possible but you'd need to go talk to the link, or a resource-level link for the "full picture" to know more.